common/djangoapps/student/models.py

```diff
@@ -1443,10 +1453,13 @@ def course_overview(self):
             become stale.
        """
         if not self._course_overview:
-            try:
-                self._course_overview = CourseOverview.get_from_id(self.course_id)
-            except (CourseOverview.DoesNotExist, IOError):
-                self._course_overview = None
+            if self.batch_course_overview:
+                self._course_overview = self.batch_course_overview
```

Comment:
One of the things that `CourseOverview.get_from_id()` does is to check the version of the `CourseOverview` data and force it to be regenerated if the existing data is from an earlier schema (i.e. doesn't have the latest fields in it). This bypasses the check, which might cause problems when we roll out code updates.

It might be that we need to just push more of this logic into `CourseOverview`, and give it a `get_many_from_ids()` method, which could then in turn be called by `enrollments_for_user()`